File: NTime/BaseCore/DataBase/Entities/ExtraColumnValue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BaseCore.DataBase
{
    public class ExtraColumnValue : IEntityId
    {
        public int Id { get; set; }
        public int PlayerId { get; set; }
        public virtual Player Player { get; set; }
        public int ColumnId { get; set; }
        public virtual ExtraColumn Column { get; set; }
        public string CustomValue { get; set; }
        public int? LookupId { get; set; }
        public virtual ExtraColumnValue Lookup { get; set; }
    }
}

File: NTime/Server/ntimeAngular/src/app/Models/ExtraColumns/ExtraColumnValue.ts
import { IExtraColumnValue } from './IExtraColumnValue';

export class ExtraColumnValue implements IExtraColumnValue {
  public Id: number;
  public PlayerId: number;
  public ColumnId: number;
  public CustomValue: string;
  public LookupId: number;

  constructor(columnId?: number) {
    if (columnId) {
      this.ColumnId = columnId;
    }
  }

  copyDataFromDto(dto: IExtraColumnValue): ExtraColumnValue {
    this.Id = dto.Id;
    this.PlayerId = dto.PlayerId;
    this.ColumnId = dto.ColumnId;
    this.CustomValue = dto.CustomValue;
    this.LookupId = dto.LookupId;
    return this;
  }
}
